lms/djangoapps/commerce/signals.py

```diff
@@ -69,78 +69,70 @@ def get_request_user():
     return getattr(request, 'user', None)
 
 
-def refund_seat(course_enrollment, request_user):
+def refund_seat(course_enrollment):
     """
-    Attempt to initiate a refund for any orders associated with the seat being
-    unenrolled, using the commerce service.
+    Attempt to initiate a refund for any orders associated with the seat being unenrolled, using the commerce service.
 
     Arguments:
         course_enrollment (CourseEnrollment): a student enrollment
-        request_user: the user as whom to authenticate to the commerce service
-            when attempting to initiate the refund.
 
     Returns:
         A list of the external service's IDs for any refunds that were initiated
             (may be empty).
 
     Raises:
-        exceptions.SlumberBaseException: for any unhandled HTTP error during
-            communication with the commerce service.
-        exceptions.Timeout: if the attempt to reach the commerce service timed
-            out.
-
+        exceptions.SlumberBaseException: for any unhandled HTTP error during communication with the E-Commerce Service.
+        exceptions.Timeout: if the attempt to reach the commerce service timed out.
     """
+    User = get_user_model()  # pylint:disable=invalid-name
     course_key_str = unicode(course_enrollment.course_id)
-    unenrolled_user = course_enrollment.user
+    enrollee = course_enrollment.user
 
-    try:
-        refund_ids = ecommerce_api_client(request_user or unenrolled_user).refunds.post(
-            {'course_id': course_key_str, 'username': unenrolled_user.username}
-        )
-    except HttpClientError, exc:
-        if exc.response.status_code == 403 and request_user != unenrolled_user:
-            # this is a known limitation; commerce service does not presently
-            # support the case of a non-superusers initiating a refund on
-            # behalf of another user.
-            log.warning("User [%s] was not authorized to initiate a refund for user [%s] "
-                        "upon unenrollment from course [%s]", request_user.id, unenrolled_user.id, course_key_str)
-            return []
-        else:
-            # no other error is anticipated, so re-raise the Exception
-            raise exc
+    service_user = User.objects.get(username=settings.ECOMMERCE_SERVICE_WORKER_USERNAME)
+    api_client = ecommerce_api_client(service_user)
+
+    log.info('Attempting to create a refund for user [%s], course [%s]...', enrollee.id, course_key_str)
+
+    refund_ids = api_client.refunds.post({'course_id': course_key_str, 'username': enrollee.username})
```

Comment:
Maybe we should talk about this offline, but why not include the process action in this POST to avoid the need for multiple calls?